### ISSUE TITLE
${var:worst} and UNKNOWN vs OK

### ISSUE THREAD

**Issue by kimheino** [munin-monitoring/munin#1240]
Commits 064c0cb224178db and fa01086edc3566a add `worst` to emails and logs, which is great improvement.

Currently worst levels are:

UNKNOWN < OK < WARNING < CRITICAL

That is, email/log says OK if some plugin returns UNKNOWN.

I would expect that levels are:

OK < UNKNOWN < WARNING < CRITICAL

Everything is not OK if some plugin fails to return data. It can indicate some service being down, which can be critical error.

**Comment by sumpfralle** [comment]
> Currently worst levels are:
> UNKNOWN < OK < WARNING < CRITICAL

Are you sure, that this is the real order of preference? Where did you notice this?

From reading the code I understood, that `OK` is only used, if no other state was reached. The relevant code starts here: https://github.com/munin-monitoring/munin/blob/064c0cb224178db34664b20734f6bee3530c1508/master/lib/Munin/Master/LimitsOld.pm#L464

The state tracking is a bit hard to read, but I think, it works as it should: OK -> UNKNOWN -> WARNING -> CRITICAL.
(the `worstid` values are not ordered - `UNKNOWN` was added later with a bigger number)

**Comment by kimheino** [comment]
Try with your foo-plugin from issue #1105, change it to return U. It sends email:

```
Subject: Munin ok: risa.bbbs.net::foo

bbbs.net :: risa.bbbs.net :: title-foo
	UNKNOWNs: label-foo is unknown.
```

Subject is "ok" (2.0.50), which comes from `${var:worst}`. I would expect it to be UNKNOWN.

**Comment by sumpfralle** [comment]
> Subject is "ok" (2.0.50), which comes from ${var:worst}. I would expect it to be UNKNOWN.

Ah! I think I fixed this (without really getting it is causing real problems) in d93f632ef1b9cc67bf166834948aaa5f6a2976e5. Do you think, this is your issue?

**Comment by kimheino** [comment]
Yes, this seems to be fixed in 2.0.51.
